### DISCUSSION TITLE
Custom Collection Views

### DISCUSSION THREAD

**Discussion by DesignyourCode** [payloadcms/payload#9345] [Feature Requests & Ideas]
# Documentation Issue

Using the website demo template out of the box the nested docs does not do everything I would expect. If enabled for a collection, I would want it to automatically (or at least allow me to opt in) to a different collection view where I can see the hierarchy of pages by parent (much like WordPress does).

My understanding is that I would need to build this view custom but their are no examples of actual custom view components in the docs that show how something like this would be achieved. A lot of examples I've found online are for Payload 2.0 which makes it different to know what will and won't work and most I have tried has resulted in errors.

## Additional Details

I think the website example template should do this out of the box. Pages should be hierarchical, nav in the header component should have nested menu items, and the routing in Next should be set up to support this across the site.

In my experience every site has nested pages and having this in and ready to go would be a game changer for getting started on a site quickly.

**Comment by kendelljoseph** [comment]
@DesignyourCode a custom view for nested docs would be cool!  It's something we have heard mentioned before.  For now, the best approach would be to create a custom view, possibly starting at the DefaultTemplate.

Let's continue this conversation going as a discussion as this would be a brand new feature.

**Comment by italodr** [comment]
I've created a plugin on `beta.128` -> https://github.com/italodr/payload-plugin-tree-list
But, on `beta.130` they changed completely the way they load views, using rsc. 🫠
I was also wondering if there was any documentation on v3 to help me refactor this plugin.

**Reply by DesignyourCode** [reply]
This plugin looks amazing. Did you get anywhere with making it working on the latest Payload version?

**Reply by italodr** [reply]
Thanks! Payload 3 is installed in the [dev folder inside the plugin](https://github.com/italodr/payload-plugin-tree-list/tree/main/dev)
I updated the code recently. The problem is that there's no method to overwrite just the table of the default list view. So it is necessary to overwrite the whole page, and reproduce everything. And it is not finished yet.
I started a [discussion](https://github.com/payloadcms/payload/discussions/9442) about it, but I will probably open it on Discord

**Reply by polynook** [reply]
The plugin indeed looks amazing. What would take it over the top is drag and drop reordering functionality.

**Comment by pbcitjugel** [comment]
It looks fantastic. Any news on this topic?
This view would make a big difference for a Web CMS!
I can't wait to have a tree view like this.
I think Wagtail is a great place to get inspiration for menu navigation.

**Comment by CarloBu** [comment]
@DesignyourCode, I was looking for a page-tree view as well. I built a page-tree UX/UI plugin for my clients - it might be useful for you too.
<p align="center">
  <img src="https://github.com/WLF-Studios/plugin-nested-docs-page-tree/raw/main/assets/page-tree.gif" alt="Page Tree Demo" />
</p>

[plugin-nested-docs-page-tree](https://github.com/WLF-Studios/plugin-nested-docs-page-tree)
